config/initializers/o_data.rb

```diff
@@ -3,4 +3,27 @@
 module OData
   NAMESPACE = "NEMO"
   BASE_PATH = "/odata/v1"
+
+  # User-friendly, doesn't need to be URL safe.
+  def self.responses_name(form_id, form_name)
+    if Settings.use_data_factory_slugs.present?
+      responses_slug(form_id, form_name)
+    else
+      "Responses: #{form_name}"
+    end
+  end
+
+  # URL safe, doesn't need to be user-friendly.
+  def self.responses_url(form_id, form_name)
+    if Settings.use_data_factory_slugs.present?
+      responses_slug(form_id, form_name)
+    else
+      "Responses-#{form_id}"
+    end
+  end
+
+  # Both user-friendly-ish and URL safe.
+  def self.responses_slug(form_id, form_name)
+    "Responses-#{form_name.gsub(/[^a-z1-9]/i, '-')}-#{form_id}"
+  end
```

Comment:
Seems like this file should be in the `lib` directory, not in initializers.

Or better yet, it seems like it really should be a class that has a `form` instance variable and wraps Form to provide this functionality. The way it's written now is not very OO. I realize this may be temporary but it also might not be.

Reply:
Hah, I didn't even think where to put them when I made this fix... Thanks for pointing it out.

I moved them to `form_decorator` which doesn't seem quite right (since they're OData specific functions rather than generic form behavior) but maybe it's okay since all the other decorators are named after models too. The code certainly reads simpler this way than before. Further suggestions welcome.

Reply:
By the way this initializer was written this way in the first place because it's used by `routes.rb` which needs it right away on startup. The new methods shouldn't have gone there though.

Reply:
Yeah I think you could use a decorator but it should be a different one, not the main one. Any time you have a bunch of prefixed methods like that in a class it's often a smell/suggestion for a new class.